File: microstartercli/src/main/resources/micronaut/security/basic/java/mongoReactive/event/LoginEventPublisher.java
package ${securityPackage}.event;

import ${securityPackage}.domains.LoginEvent;
import io.micronaut.context.event.ApplicationEventPublisher;
import io.micronaut.scheduling.annotation.Async;
import jakarta.inject.Inject;
import jakarta.inject.Singleton;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Singleton
public class LoginEventPublisher
{
    private static final Logger log = LoggerFactory.getLogger(LoginEventPublisher.class);
    @Inject
    ApplicationEventPublisher eventPublisher;

    @Async
    public void publishEvent(LoginEvent loginEvent) {
        log.info("Publishing login event: {}", loginEvent);

        eventPublisher.publishEvent(loginEvent);
    }
}

File: Code Examples/JWT/Kotlin/JWTKotlinMongo/src/main/kotlin/io/hashimati/security/event/LoginEventPublisher.kt
package io.hashimati.security.event

import io.hashimati.security.domains.LoginEvent
import io.micronaut.context.event.ApplicationEventPublisher
import io.micronaut.scheduling.annotation.Async
import jakarta.inject.Inject
import jakarta.inject.Singleton
import org.slf4j.LoggerFactory

@Singleton
class LoginEventPublisher {

    @Inject
    internal var eventPublisher: ApplicationEventPublisher<LoginEvent>? = null

    fun publishEvent(loginEvent: LoginEvent) {
        log.info("Publishing login event: {}", loginEvent)
        eventPublisher!!.publishEvent(loginEvent )
   }

    companion object {
        private val log = LoggerFactory.getLogger(LoginEventPublisher::class.java)
    }
}